Compilation failure with gcc 16

Thank you for taking the time to submit an issue!

## Background information

Fedora 44 has gcc 16.  openmpi fails to build with:
```
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../../opal/include -I../../../ompi/include -I../../../oshmem/include -I../../../ompi/mpiext/cuda/c -I../../../ompi/mpiext/rocm/c -DOMPI_LIBMPI_NAME=\"mpi\" -iquote../../.. -I/usr/include/pmix -DNDEBUG -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -finline-functions -mcx16 -MT base/memheap_base_frame.lo -MD -MP -MF base/.deps/memheap_base_frame.Tpo -c base/memheap_base_frame.c  -fPIC -DPIC -o base/.libs/memheap_base_frame.o
base/memheap_base_frame.c:38:1: warning: braces around scalar initializer
   38 | mca_memheap_map_t mca_memheap_base_map = {{{{0}}}};
      | ^~~~~~~~~~~~~~~~~
base/memheap_base_frame.c:38:1: note: (near initialization for ‘mca_memheap_base_map.mem_segs’)
base/memheap_base_frame.c:38:1: error: braces around scalar initializer
base/memheap_base_frame.c:38:1: note: (near initialization for ‘mca_memheap_base_map.mem_segs’)
base/memheap_base_frame.c:38:1: error: braces around scalar initializer
base/memheap_base_frame.c:38:1: note: (near initialization for ‘mca_memheap_base_map.mem_segs’)
make[2]: *** [Makefile:1532: base/memheap_base_frame.lo] Error 1
```

### What version of Open MPI are you using? (e.g., v4.1.6, v5.0.1, git branch name and hash, etc.)
5.0.9

I don't see any change in this file in current git main.

--- a/oshmem/mca/memheap/base/memheap_base_frame.c
+++ b/oshmem/mca/memheap/base/memheap_base_frame.c
@@ -33,9 +33,9 @@
 
 int mca_memheap_base_output = -1;
 int mca_memheap_base_key_exchange = 1;
-opal_list_t mca_memheap_base_components_opened = {{0}};
+opal_list_t mca_memheap_base_components_opened = {};
 int mca_memheap_base_already_opened = 0;
-mca_memheap_map_t mca_memheap_base_map = {{{{0}}}};
+mca_memheap_map_t mca_memheap_base_map = {};
 int mca_memheap_num_segments_warn = 32;
 
 static int mca_memheap_base_register(mca_base_register_flag_t flags)

